### PR TITLE
add /tasks/{taskType}/{tableNameWithType}/debug API

### PR DESCRIPTION
**Pull request by zhtaoxiang**
## What Does This PR Do
Add /tasks/{taskType}/{tableNameWithType}/debug API so that we can get a task debug info for the given table. This makes debugging a specific table's problem easier.

## Test
Test by setting up a Pinot cluster locally.

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTaskRestletResource.java [hunk 1]
```diff
@@ -215,6 +215,18 @@ public Map<String, PinotHelixTaskResourceManager.TaskDebugInfo> getTasksDebugInf
...
+  @GET
+  @Path("/tasks/{taskType}/{tableNameWithType}/debug")
+  @ApiOperation("Fetch information for the given task type and table")
+  public Map<String, PinotHelixTaskResourceManager.TaskDebugInfo> getTasksDebugInfo(
+      @ApiParam(value = "Task type", required = true) @PathParam("taskType") String taskType,
+      @ApiParam(value = "Table name with type", required = true) @PathParam("tableNameWithType")
+          String tableNameWithType,
+      @ApiParam(value = "verbosity (By default, prints for running and error tasks. Value of >0 prints for all tasks)")
+      @DefaultValue("0") @QueryParam("verbosity") int verbosity) {
+    return _pinotHelixTaskResourceManager.getTasksDebugInfoByTable(taskType, tableNameWithType, verbosity);
```
Comment (npawar) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTaskRestletResource.java:227]: I tried this out with RealtimeQuickstartWithMinion, and saw it return 3 COMPLETED tasks, for both 0 and 1 verbosity. Only difference was level of detail. Is this `By default, prints for running and error tasks. Value of >0 prints for all tasks)` incorrect or does the impl need some change?

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotHelixTaskResourceManager.java [hunk 2]
```diff
@@ -531,6 +531,48 @@ public synchronized Map<String, TaskDebugInfo> getTasksDebugInfo(String taskType
...
+      // Iterate through all task configs associated with this task name
+      for (PinotTaskConfig taskConfig : getTaskConfigs(pinotTaskName)) {
+        Map<String, String> pinotConfigs = taskConfig.getConfigs();
+
+        // Filter task configs that matches this table name
+        if (pinotConfigs != null) {
+          String tableNameConfig = pinotConfigs.get(TABLE_NAME);
+          if (tableNameConfig != null && tableNameConfig.equals(tableNameWithType)) {
+            // Found a match. Add task debug info to the result
+            // TODO: we may want to filter out debug info that does not belong to the given table.
```
Comment (npawar) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotHelixTaskResourceManager.java:566]: didn't follow this TODO. why would we get info from `getTaskDebugInfo` that's not for this table?

**Comment by npawar** [comment]
> ## What Does This PR Do
> Add /tasks/{taskType}/{tableNameWithType}/debug API so that we can get a task debug info for the given table. This makes debugging a specific table's problem easier.
> 
> ## Test
> Test by setting up a Pinot cluster locally.

Mind adding a sample response here?

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTaskRestletResource.java (hunk 1 above)
Comment (zhtaoxiang) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTaskRestletResource.java:227]: simply copied and pasted the existing docs 😄  . fixed this one and existing ones.

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotHelixTaskResourceManager.java (hunk 2 above)
Comment (zhtaoxiang) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotHelixTaskResourceManager.java:566]: updated the TODO.

**Comment by zhtaoxiang** [comment]
`curl -X GET "http://localhost:18998/tasks/FileIngestionTask/fileIngestionTaskTestTable_AdhocExecute_OFFLINE/debug?verbosity=0" -H "accept: application/json"`

```
{
  "Task_FileIngestionTask_fileIngestionTaskTestTable_AdhocExecute_OFFLINE_1a36671a-c451-41b1-94a0-86cfdf6dd67e": {
    "taskState": "COMPLETED",
    "subtaskCount": {
      "total": 2,
      "completed": 2,
      "running": 0,
      "waiting": 0,
      "error": 0,
      "unknown": 0
    },
    "startTime": "2022-06-21 19:54:35 PDT",
    "executionStartTime": "2022-06-21 19:54:35 PDT"
  },
  "Task_FileIngestionTask_fileIngestionTaskTestTable_AdhocExecute_OFFLINE_2c9e0820-1a0b-4bec-a17d-1d06403727f5": {
    "taskState": "COMPLETED",
    "subtaskCount": {
      "total": 2,
      "completed": 2,
      "running": 0,
      "waiting": 0,
      "error": 0,
      "unknown": 0
    },
    "startTime": "2022-06-21 19:55:03 PDT",
    "executionStartTime": "2022-06-21 19:55:03 PDT"
  },
  "Task_FileIngestionTask_fileIngestionTaskTestTable_AdhocExecute_OFFLINE_b7294c35-45aa-4450-ae5a-82eee4a4ef49": {
    "taskState": "COMPLETED",
    "subtaskCount": {
      "total": 2,
      "completed": 2,
      "running": 0,
      "waiting": 0,
      "error": 0,
      "unknown": 0
    },
    "startTime": "2022-06-21 19:54:50 PDT",
    "executionStartTime": "2022-06-21 19:54:50 PDT"
  }
}
```

**Comment by zhtaoxiang** [comment]
Sample response when verbosity = 1

`curl -X GET "http://localhost:18998/tasks/FileIngestionTask/fileIngestionTaskTestTable_AdhocExecute_OFFLINE/debug?verbosity=1" -H "accept: application/json"`

```
{
  "Task_FileIngestionTask_fileIngestionTaskTestTable_AdhocExecute_OFFLINE_1a36671a-c451-41b1-94a0-86cfdf6dd67e": {
    "taskState": "COMPLETED",
    "subtaskCount": {
      "total": 2,
      "completed": 2,
      "running": 0,
      "waiting": 0,
      "error": 0,
      "unknown": 0
    },
    "startTime": "2022-06-21 19:54:35 PDT",
    "executionStartTime": "2022-06-21 19:54:35 PDT",
    "subtaskInfos": [
      {
        "taskId": "Task_FileIngestionTask_fileIngestionTaskTestTable_AdhocExecute_OFFLINE_1a36671a-c451-41b1-94a0-86cfdf6dd67e_1",
        "state": "COMPLETED",
        "startTime": "2022-06-21 19:54:35 PDT",
        "finishTime": "2022-06-21 19:54:39 PDT",
        "participant": "Minion_127.0.0.1_9514",
        "info": "Succeeded",
        "taskConfig": {
          "taskType": "FileIngestionTask",
          "configs": {
            "recordReader.configClassName": "org.apache.pinot.plugin.inputformat.csv.CSVRecordReaderConfig",
            "maxNumRecordsPerSegment": "1000",
            "inputDirURI": "file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/",
            "segmentNamePrefix": "fileIngestionTaskTestTable_AdhocExecute",
            "tableMaxNumTasks": "2",
            "inputFormat": "csv",
            "sequenceId": "FileIngestionTask_1655866466031_1",
            "segmentNamePostfix": "FileIngestionTask_1655866466031_1",
            "tableName": "fileIngestionTaskTestTable_AdhocExecute_OFFLINE",
            "recordReader.className": "org.apache.pinot.plugin.inputformat.csv.CSVRecordReader",
            "checkpointTTLSec": "0",
            "input.data.file.uri": "file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_00003.csv,file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_00004.csv,file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_00005.csv",
            "taskMaxNumFiles": "3",
            "uploadURL": "http://localhost:18998/segments",
            "push.controllerUri": "http://localhost:18998"
          }
        }
      },
      {
        "taskId": "Task_FileIngestionTask_fileIngestionTaskTestTable_AdhocExecute_OFFLINE_1a36671a-c451-41b1-94a0-86cfdf6dd67e_0",
        "state": "COMPLETED",
        "startTime": "2022-06-21 19:54:35 PDT",
        "finishTime": "2022-06-21 19:54:38 PDT",
        "participant": "Minion_127.0.0.1_9514",
        "info": "Succeeded",
        "taskConfig": {
          "taskType": "FileIngestionTask",
          "configs": {
            "recordReader.configClassName": "org.apache.pinot.plugin.inputformat.csv.CSVRecordReaderConfig",
            "maxNumRecordsPerSegment": "1000",
            "inputDirURI": "file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/",
            "segmentNamePrefix": "fileIngestionTaskTestTable_AdhocExecute",
            "tableMaxNumTasks": "2",
            "inputFormat": "csv",
            "sequenceId": "FileIngestionTask_1655866466031_0",
            "segmentNamePostfix": "FileIngestionTask_1655866466031_0",
            "tableName": "fileIngestionTaskTestTable_AdhocExecute_OFFLINE",
            "recordReader.className": "org.apache.pinot.plugin.inputformat.csv.CSVRecordReader",
            "checkpointTTLSec": "0",
            "input.data.file.uri": "file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_00000.csv,file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_00001.csv,file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_00002.csv",
            "taskMaxNumFiles": "3",
            "uploadURL": "http://localhost:18998/segments",
            "push.controllerUri": "http://localhost:18998"
          }
        }
      }
    ]
  },
  "Task_FileIngestionTask_fileIngestionTaskTestTable_AdhocExecute_OFFLINE_2c9e0820-1a0b-4bec-a17d-1d06403727f5": {
    "taskState": "COMPLETED",
    "subtaskCount": {
      "total": 2,
      "completed": 2,
      "running": 0,
      "waiting": 0,
      "error": 0,
      "unknown": 0
    },
    "startTime": "2022-06-21 19:55:03 PDT",
    "executionStartTime": "2022-06-21 19:55:03 PDT",
    "subtaskInfos": [
      {
        "taskId": "Task_FileIngestionTask_fileIngestionTaskTestTable_AdhocExecute_OFFLINE_2c9e0820-1a0b-4bec-a17d-1d06403727f5_1",
        "state": "COMPLETED",
        "startTime": "2022-06-21 19:55:03 PDT",
        "finishTime": "2022-06-21 19:55:03 PDT",
        "participant": "Minion_127.0.0.1_9514",
        "info": "Succeeded",
        "taskConfig": {
          "taskType": "FileIngestionTask",
          "configs": {
            "recordReader.configClassName": "org.apache.pinot.plugin.inputformat.csv.CSVRecordReaderConfig",
            "maxNumRecordsPerSegment": "1000",
            "inputDirURI": "file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/",
            "segmentNamePrefix": "fileIngestionTaskTestTable_AdhocExecute",
            "tableMaxNumTasks": "2",
            "inputFormat": "csv",
            "sequenceId": "FileIngestionTask_1655866492891_1",
            "segmentNamePostfix": "FileIngestionTask_1655866492891_1",
            "tableName": "fileIngestionTaskTestTable_AdhocExecute_OFFLINE",
            "recordReader.className": "org.apache.pinot.plugin.inputformat.csv.CSVRecordReader",
            "checkpointTTLSec": "0",
            "input.data.file.uri": "file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_Dir1/emptyFile_00002.csv,file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_Dir1/emptyFile_00003.csv,file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_Dir1/emptyFile_00004.csv",
            "taskMaxNumFiles": "3",
            "uploadURL": "http://localhost:18998/segments",
            "push.controllerUri": "http://localhost:18998"
          }
        }
      },
      {
        "taskId": "Task_FileIngestionTask_fileIngestionTaskTestTable_AdhocExecute_OFFLINE_2c9e0820-1a0b-4bec-a17d-1d06403727f5_0",
        "state": "COMPLETED",
        "startTime": "2022-06-21 19:55:03 PDT",
        "finishTime": "2022-06-21 19:55:04 PDT",
        "participant": "Minion_127.0.0.1_9514",
        "info": "Succeeded",
        "taskConfig": {
          "taskType": "FileIngestionTask",
          "configs": {
            "recordReader.configClassName": "org.apache.pinot.plugin.inputformat.csv.CSVRecordReaderConfig",
            "maxNumRecordsPerSegment": "1000",
            "inputDirURI": "file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/",
            "segmentNamePrefix": "fileIngestionTaskTestTable_AdhocExecute",
            "tableMaxNumTasks": "2",
            "inputFormat": "csv",
            "sequenceId": "FileIngestionTask_1655866492891_0",
            "segmentNamePostfix": "FileIngestionTask_1655866492891_0",
            "tableName": "fileIngestionTaskTestTable_AdhocExecute_OFFLINE",
            "recordReader.className": "org.apache.pinot.plugin.inputformat.csv.CSVRecordReader",
            "checkpointTTLSec": "0",
            "input.data.file.uri": "file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_00012.csv,file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_Dir1/emptyFile_00000.csv,file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_Dir1/emptyFile_00001.csv",
            "taskMaxNumFiles": "3",
            "uploadURL": "http://localhost:18998/segments",
            "push.controllerUri": "http://localhost:18998"
          }
        }
      }
    ]
  },
  "Task_FileIngestionTask_fileIngestionTaskTestTable_AdhocExecute_OFFLINE_b7294c35-45aa-4450-ae5a-82eee4a4ef49": {
    "taskState": "COMPLETED",
    "subtaskCount": {
      "total": 2,
      "completed": 2,
      "running": 0,
      "waiting": 0,
      "error": 0,
      "unknown": 0
    },
    "startTime": "2022-06-21 19:54:50 PDT",
    "executionStartTime": "2022-06-21 19:54:50 PDT",
    "subtaskInfos": [
      {
        "taskId": "Task_FileIngestionTask_fileIngestionTaskTestTable_AdhocExecute_OFFLINE_b7294c35-45aa-4450-ae5a-82eee4a4ef49_0",
        "state": "COMPLETED",
        "startTime": "2022-06-21 19:54:50 PDT",
        "finishTime": "2022-06-21 19:54:51 PDT",
        "participant": "Minion_127.0.0.1_9514",
        "info": "Succeeded",
        "taskConfig": {
          "taskType": "FileIngestionTask",
          "configs": {
            "recordReader.configClassName": "org.apache.pinot.plugin.inputformat.csv.CSVRecordReaderConfig",
            "maxNumRecordsPerSegment": "1000",
            "inputDirURI": "file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/",
            "segmentNamePrefix": "fileIngestionTaskTestTable_AdhocExecute",
            "tableMaxNumTasks": "2",
            "inputFormat": "csv",
            "sequenceId": "FileIngestionTask_1655866480047_0",
            "segmentNamePostfix": "FileIngestionTask_1655866480047_0",
            "tableName": "fileIngestionTaskTestTable_AdhocExecute_OFFLINE",
            "recordReader.className": "org.apache.pinot.plugin.inputformat.csv.CSVRecordReader",
            "checkpointTTLSec": "0",
            "input.data.file.uri": "file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_00006.csv,file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_00007.csv,file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_00008.csv",
            "taskMaxNumFiles": "3",
            "uploadURL": "http://localhost:18998/segments",
            "push.controllerUri": "http://localhost:18998"
          }
        }
      },
      {
        "taskId": "Task_FileIngestionTask_fileIngestionTaskTestTable_AdhocExecute_OFFLINE_b7294c35-45aa-4450-ae5a-82eee4a4ef49_1",
        "state": "COMPLETED",
        "startTime": "2022-06-21 19:54:50 PDT",
        "finishTime": "2022-06-21 19:54:52 PDT",
        "participant": "Minion_127.0.0.1_9514",
        "info": "Succeeded",
        "taskConfig": {
          "taskType": "FileIngestionTask",
          "configs": {
            "recordReader.configClassName": "org.apache.pinot.plugin.inputformat.csv.CSVRecordReaderConfig",
            "maxNumRecordsPerSegment": "1000",
            "inputDirURI": "file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/",
            "segmentNamePrefix": "fileIngestionTaskTestTable_AdhocExecute",
            "tableMaxNumTasks": "2",
            "inputFormat": "csv",
            "sequenceId": "FileIngestionTask_1655866480047_1",
            "segmentNamePostfix": "FileIngestionTask_1655866480047_1",
            "tableName": "fileIngestionTaskTestTable_AdhocExecute_OFFLINE",
            "recordReader.className": "org.apache.pinot.plugin.inputformat.csv.CSVRecordReader",
            "checkpointTTLSec": "0",
            "input.data.file.uri": "file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_00009.csv,file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_00010.csv,file:/var/folders/kd/4c8chyjn6kdcy7bmv69x6bc40000gn/T/FileIngestionTaskIntegrationTest/fileIngestionTaskTestTable_AdhocExecute/tempFile_00011.csv",
            "taskMaxNumFiles": "3",
            "uploadURL": "http://localhost:18998/segments",
            "push.controllerUri": "http://localhost:18998"
          }
        }
      }
    ]
  }
}
```

**Review by npawar — APPROVED** [review]

**Pull request merged**